IPython/terminal/embed.py

```diff
@@ -83,6 +96,8 @@ def __init__(self, **kw):
 
         super(InteractiveShellEmbed,self).__init__(**kw)
 
+        frame = sys._getframe(InteractiveShellEmbed._stack_frames)
+        self._call_location_id = '%s:%s' % (frame.f_code.co_filename, frame.f_lineno)
```

Comment:
I'm in favor of location, and the issues with giving the caller is that the caller can likely change, and still be in the same location. I can add a `call_location` kwarg that take precedence on my logic.